### DISCUSSION TITLE
Can sandbox mode be played continually in the background?

### DISCUSSION THREAD

**Discussion by CoffeeAddictAnonymous** [mars-sim/mars-sim#324] [Q&A] [answered]
Without any input, just to see things (evolve) and progress?

I ask, because I left the game running over night and each colony was left with 1-2 people and everyone else died (playing github version). Is this just a normal thing of living on Mars?

**Comment by bevans2000** [comment, marked as answer]
No there is a bug in the Airlock logic #305 which means Settlers randomly get stuck.

**Comment by Urwumpe** [comment]
Sadly not, first of all because of the bug mentioned, but also the Settler intelligence isn't that far already to care about their own survival. 

But yes, that is also the preferred test case for me: Starting MSP in the morning and check at lunch how far it came. I also hope that it would eventually be able to run for a full martian year again.

**Comment by mokun** [comment]
Currently, the airlock bug is the show-stopper.

Besides fixing this bug, another crucial thing I can think of is re-enabling the ability of settlers' enduring effort of expanding the human settlement.

To do so, we will need to fix the bug in starting the mission of setting new foundations and framing new buildings.

Feel free to add anything you may have observed.

**Reply by IBuildVans** [reply]
From the discussion, it looks like the airlock bug was fixed?   If so, will the sim run on its own now without intervention?   Will the bases expand automatically?

**Reply by mokun** [reply]
@IBuildVans, yes i'd say it's fixed when settlers are ingressing and egressing the settlement.

I did seem to witness an edge case a few days ago in which a settler gets trapped in a vehicular airlock and was too exhausted to get in or get out. I will need to take a closer look. 

If you see anything unusual, do post your screenshots here. Thanks !

**Reply by IBuildVans** [reply]
I'm not sure this is related to the airlock issue but I started a new run of the sim yesterday and let it run overnight.   When I checked today, I found that one colony had 4 of 4  deceased and another colony had 2 of 4 deceased.   I'm runnign the headless version so I'm hoping I can go through the logs and find what happened.

**Reply by mokun** [reply]
>  found that one colony had 4 of 4 deceased and another colony had 2 of 4 deceased.

The Event Tab in `Monitor Tool` would tell you what has transpired. 

There's a good chance that settlers are out on a mission and an accident happens and resources dwindled to the point that they ran out of it.

mars-sim does have emergency mission for handling rescue operations. But things can still get haywire that it's too late.

Btw, it has been said mars-sim is designed with too much automation since day one. 

As a result, it doesn't requires much of player interaction. This is both the strength and weakness of mars-sim.

**Reply by mokun** [reply]
I just found out a person can get stuck at the pre-breathing phase of egressing an vehicular airlock while on a mission out there somewhere.

![image](https://github.com/mars-sim/mars-sim/assets/1168584/90b27146-e42f-4297-9dc2-c7eb0e8baa1e)

![image](https://github.com/mars-sim/mars-sim/assets/1168584/1b5cb2c3-a0b0-47aa-9eab-0dc862e7f02a)

Fixing it now.

**Reply by IBuildVans** [reply]
Thank you again.   I am running th console(remote) version of the sim so I miss out on a lot of the GUI tools.   I like the "automation" of the colony just to see what happens.  However I would also enjoy being able to have more input.   Maybe that is possible  with the GUI version, im not sure.

**Reply by IBuildVans** [reply]
![alldead](https://github.com/mars-sim/mars-sim/assets/51859034/1f7ed09f-809b-4035-89f5-615c8b929744)


Why would colonists die while doing maintenance?

**Reply by mokun** [reply]
Ah, I can see the death reporting is incomplete.

I suspect that for some reasons they are unable to come back in via the airlock and get stuck during ingress.

I'll modify `StatusCommand` class to show the hunger, energy, thirst, stress, fatigue and performance of a person at the time of death. It will reveal a bit more on the cause of death or at least provide some more rational reasons as to why he/she died.

Then I'll compile a new jar for you to test run it.

**Reply by mokun** [reply]
Btw, the cause of death won't be determined until after the body is retrieved and another person will perform a post-mortem exam. 

When this exam is done, it will post any underlining medical issue (if available).

**Comment by IBuildVans** [comment]
Thank you!  I will be happy to try out a new build!  If everyone in the
colony is dead, can I force someone from another settlement to go and
retrieve the bodies?  Or is there any other way to do that?

**Reply by mokun** [reply]
>  can I force someone from another settlement to go and retrieve the bodies?

We haven't considered this particular case yet. 

It may or may not be realistic if the nearest settlement is over 1000 km one-way. 

By the time a rover arrives, it would be run out of fuel.

There's no guarantee that there's any methanol (used by rovers) left in that settlement to replenish. 

But I suppose if a settlement is on the verge of collapse, it will ask for the sponsoring agency for help. Then it's possible that an international rescue effort be cemented...

So feel free to join in other discussions such as [this](https://github.com/mars-sim/mars-sim/issues/1095) with regard to what political and diplomatic relationship to model among the sponsors, among the nation states, and among the settlements, etc.

**Reply by IBuildVans** [reply]
Yes, that makes sense about the distance.  I think in my example it would be more of a recovery as opposed to a rescue.   Maybe something could be added where as a final action, the colonists can activate an emergency beacon.  Like if they know they are stuck and running out of oxygent, etc.    Maybe a different settlement could detect it and if they couldnt do anything else maybe they could relay the details back to earth.  Just throwing out ideas. :)

**Reply by mokun** [reply]
Sure. Rovers already have emergency beacon and await for rescue. 

I suppose there's very little wiggle room for anyone to do anything or go anywhere if accidents happen and/or food or oxygen runs out. 

We could in future model the idea of the airlift. 

We added cargo drones in this [ticket](https://github.com/mars-sim/mars-sim/issues/361).

A remotely piloted, two-seater transport **Passenger Drone** could become the best hope for ferrying settlers quickly from settlement to settlement.

**Comment by IBuildVans** [comment]
This is from the same run as the other colony where everyone is dead.   The simulation has been running for about 2 real time days now.  I found this when I checked on them today...
![alldead-2](https://github.com/mars-sim/mars-sim/assets/51859034/67cf7dfb-088f-499f-933f-38061120ad4d)

**Reply by mokun** [reply]
Looks like we'll need to look at the general health of this settlement. May be it ran out of oxygen, food, or water ? 

Ideally speaking, if they knows the settlement is doomed, they won't do anything that's inconsequential. They may be writing emails or connecting with people on earth when the resources run out... We could deepen these aspects of simulation in a more realistic manner (but not becoming a pessimist for going to Mars :D) .

**Comment by IBuildVans** [comment]
It looks like I will need to restart.   I checked the other colonies and
everyone is dead.   I guess the sim has essentially ended at this point.  .

**Reply by mokun** [reply]
@IBuildVans @CoffeeAddictAnonymous 

We'll be releasing 3.7.0 soon. Settlers can survive much longer now. 

In 2 of my simulations I tested the other day, they go beyond 60-80 sols without major issues that would require my intervention.

**Comment by IBuildVans** [comment]
Awesome, thanks for the update!   I will absolutely try out 3.7.0 when it
is ready!
